sstables/sstable_directory.hh

```diff
@@ -75,7 +75,7 @@ public:
     public:
         virtual future<> process(sstable_directory& directory, process_flags flags) = 0;
         virtual future<> commit() = 0;
-        virtual future<> garbage_collect() = 0;
+        virtual future<> garbage_collect(storage&) = 0;
```

Comment:
a bit out of scope: do you have further plans for commit()? cannot we consolide garbage_collect() and commit()? it seems commit() in fs mode just wipes files that happened to have temporary toc (= unsealed), whereas garbage_collect() is wiping files that are part of atomic deletion. both are garbage to be collected.

Reply:
After https://github.com/scylladb/scylladb/pull/15318#discussion_r1321490712 I probably do have plans for it -- to remove all garbage TOCs. But the more I think of it the more I dislike it :( So likely I don't have plans for it